src/librustc_mir/pretty.rs

```diff
@@ -66,9 +66,15 @@ pub fn dump_mir<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
         _ => String::new()
     };
 
+    let mut file_path = String::new();
+    if let Some(ref file_dir) = tcx.sess.opts.debugging_opts.dump_mir_dir {
+        file_path.push_str(file_dir);
+    };
     let file_name = format!("rustc.node{}{}.{}.{}.mir",
                             node_id, promotion_id, pass_name, disambiguator);
-    let _ = fs::File::create(&file_name).and_then(|mut file| {
+    file_path.push_str("/");
```

Comment:
Doesn't this mean that, if no path is specified, the MIR will be dumped into the root folter (`/rustc.node[...].mir`)? Using `PathBuf` can avoid that, and would be semantically more appropriate and more robust anyway.